cpp/cmake/Dependencies.cmake

```diff
@@ -16,6 +16,32 @@
 
 include(ExternalProject)
 
+##############################################################################
+# - raft - (header only) -----------------------------------------------------
+
+# Only cloned if RAFT_PATH env variable is not defined
+
+if(DEFINED ENV{RAFT_PATH})
+  message(STATUS "RAFT_PATH environment variable detected.")
+  message(STATUS "RAFT_DIR set to $ENV{RAFT_PATH}")
+  set(RAFT_DIR ENV{RAFT_PATH})
+
+else(DEFINED ENV{RAFT_PATH})
+  message(STATUS "RAFT_PATH environment variable NOT detected, cloning RAFT")
+  set(RAFT_GIT_DIR ${CMAKE_CURRENT_BINARY_DIR}/raft CACHE STRING "Path to RAFT repo")
+
+  ExternalProject_Add(raft
+    GIT_REPOSITORY    https://github.com/rapidsai/raft.git
+    GIT_TAG           a1a550a37c5fd2e9dff9ff81775a1542ba0af0d7
+    PREFIX            ${RAFT_GIT_DIR}
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND     ""
+    INSTALL_COMMAND   "")
+
+  set(RAFT_INCLUDE_DIR ${RAFT_GIT_DIR}/src/raft/cpp/include CACHE STRING "Path to RAFT repo")
```

Comment:
Do we have to add RAFT_INCLUDE_DIR to our include dirs for libcuml?

Reply:
Probably better to move this outside of this if-else block?

Reply:
Added a comment of why it needs to be there and corrected the name for consistency with the other `PROJECT_DIR` variables